### ISSUE TITLE
check appsecret_proof parameter

### ISSUE THREAD

**Issue by nbartels** [restfb/restfb#243]
check if the appsecret_proof parameter is added in request that come from a application and check the result facebook sends back

**Comment by JeanKeens** [comment]
Hi Mr. Bartels
I am currently working on publishing on a FB page with restfb.
I for the time being having a hard time on this precise issue.

I have a choice to make:
Either we contribute your project, or we fork it.
Are you opened to cross-company work on open source? If the answer's yes, then we would like to start discussions about governance, and licensing issues.
I can see a licence file mentionning a copyright @nbartels & Mark Allen (but you're the only contributor).

Working together is always considered as a success at SPVCE, plus I definitely wanted to make the offer, before making any decision, for the sake of curtesy between pairs.

feel free to contact us at dev.social.mngt.fb@spvce.net

Yours Sincerely,

Jean-Baptiste Lasselle, CEO
www.spvce.net

**Comment by revetkn** [comment]
Hi Jean-Baptiste

We are certainly open to contributions - normally this is done via a pull request.  Or, if you do not have a pull request, maybe you can create a new issue and we can take a look at it.  I am not sure that governance, authoring, and licensing come into play here, at least not until we know more about your issue.

**Comment by JeanKeens** [comment]
Hi Mark,
My issue is https://github.com/restfb/restfb/issues/243 . (Are we not commenting on issue #243 ? ...)

Status of that issue is "in-pogress". As far as I could understand here:
https://groups.google.com/forum/#!topic/restfb/mWZ90eLCh0Q, and as far as my work brought me, I
understand it is not possible to use restfb to publish messages, events etc... to a facebook page, since the FacebookClient.publish(...)  method is bugged.
Is there any alternative way to publish to Facebook pages that actually works?

Right, I was considering fixing the bug, since the fix has been in-progress for two months by now, and I since there is only one contributor, I could not expect a quick bug fix release : I'm sure @nbartels has got plenty of work to do at the office, such as all of us IT guys do.

Fine, then if I fix the bug, there two possible ways to go forward: either I push the bugfix to your project (contribute, I'm fully aware that contributing on github is making a pull request), or I fork.

Fine, so I look up the git repo, and found no licensing such as LGPL, GPL, creative commons etc...
Just copyrights (I believe, on the name "restfb" => meaning Iyou can sue me If I reslease any alike product and use the name "restfb" for that product, but I'm not a copyrights SME).

When you consider contributing to a project, governance, authoring, and licensing are the criteria for a decision. Which is why I proposed discussion on the matter.

If you want to know more about how governance, authoring, and licensing can be tightly coupled to that kind of decision, well you can read a lot about what Richard Stallman wrote and did about the subject.

And I do share Richard Stallman's vision on the subject.

Two very practical aspects  involved:

A) To seriously consider contributing, we would need a shared Continuous Integration Infrastructure (Where is restfb 's CI server? How to we get proper access credentials to it?) - I'd be very interested in helping with (re)designing that Infastructure, if helpful.
B) There is no way I will give any code for free, unless I have assurance (such as in a GPL context) that my name will persist as author fo all my contributions.
C) Of course, We're interested in restfb because it is quoted on official Facebook developers' page listing SDKs. Plus we think that there is a lot to contribute about, on restfb, which s a great opportunity.


All in all, I thought discussions by emails would be more appropriate for such governance discussions, which the reason I gave you:

Feel free to contact us at dev.social.mngt.fb@spvce.net


We're still opened. Your decision.



Yours Sincerely,

Jean-Baptiste Lasselle, CEO

www.spvce.net

**Comment by revetkn** [comment]
Hi Jean-Baptiste

This issue, #243, has been marked closed since June 18 as part of the 1.12.0 release (@nbartels - please correct if wrong).  The "in-progress" label should have been removed, I'll do that now...I think that might have been the source of confusion.  RestFB is released under the terms of the [MIT License](https://en.wikipedia.org/wiki/MIT_License) per the top of the README at https://github.com/restfb/restfb as well as on http://restfb.com.  Currently there are no plans to change the license.  As you mention, it would be good to have a CI server set up, but currently we do not have one.  So - if the issue is fixed - hopefully we do not need any additional work done :smile:  If you are still running into problems publishing with version 1.14.0, maybe post some example code and we can figure it out together!

**Comment by JeanKeens** [comment]
Hi again Mark,

Thank you so much for such a quick answer! I'm right going back to my POC work, and give feedback very very soon.

ps: About the CI server set-up, would you be interested in SPVCE 's contribution on the subject? Again, I am very much into quality managment, and SPVCE 's would be very glad & proud to communicate we have done that for restfb!

Yours sincerely

JBL

**Comment by nbartels** [comment]
Hi all,

this issue was reported some time ago, but it turned out to be a no-issue. So I added an integration test as proof of correctness. 

ATM, we have no public CI Server, but of course a private one :smile: 

It's no problem to connect the repository to travis ci for example, but it should provide an additional benefit. I really like to use cool tools, but don't like to bother a user with these details.

Cheers
Norbert

**Comment by JeanKeens** [comment]
Hi Mr. Bartels,
Alright ... All teams that publcicly expose their CI server will be glad to find out that its just something bothering users ... (*smiling high :D )...

**Comment by tunkl** [comment]
Hi, I have tried the integration test you have mentioned here. I still got the same error. Is there some kind of configuration in APP which must be done? Or some way how the token is generated so this works? Without the appsecret_proof param I can read the data, so the token should be valid.

**Comment by nbartels** [comment]
Can you provide some more information? And can you please double check the necessary inputs like access token and app secret?

I'll check the integration test, too. But at some point of time it worked :smile:

**Comment by nbartels** [comment]
So, I checked the integration test code and everything is working as expected.

Perhaps you can activate the debug logging for com.restfb.HTTP. You should see the requests and responses and perhaps you get some idea was going wrong.

**Comment by tunkl** [comment]
Hi, thanks for your help. I have tried the test again with different access_token and it worked. Seems to me that it depends on how old is the token. Is it possible?

**Comment by tunkl** [comment]
Sorry I have double checked. The problem is on my side. I have switched test application and production one. Really sorry.

**Comment by nbartels** [comment]
Okay. Then have a lot of fun using restfb...
